src/plugins/intel_cpu/src/emitters/tpp/aarch64/kernel_executors/brgemm.hpp

```diff
@@ -0,0 +1,123 @@
+
+    std::unique_ptr<snippets::KernelExecutorBase::GenericConfig> get_clone_ptr() const override {
+        return std::unique_ptr<BrgemmKernelConfig>(new BrgemmKernelConfig(*this));
+    }
+
+    dnnl_data_type_t get_dt_in0() const {
+        return get_static_params()->dt_in0;
+    }
+    dnnl_data_type_t get_dt_in1() const {
+        return get_static_params()->dt_in1;
+    }
+    dnnl::impl::cpu::aarch64::cpu_isa_t get_isa() const {
+        return m_static_params->isa;
+    }
+    libxsmm_bitfield get_static_compile_flags() const {
+        return m_static_params->m_compile_flags;
+    }
+    libxsmm_bitfield get_compile_flags() const {
+        return m_compile_flags;
+    }
+    void set_compile_flags(bool zero_beta) {
+        if (zero_beta) {
+            m_compile_flags = get_static_compile_flags() | LIBXSMM_GEMM_FLAG_BETA_0;
+        } else {
+            m_compile_flags = get_static_compile_flags();
+        }
+    }
+    bool get_prefetching_flags() const {
+        return m_static_params->m_prefetching_flags;
+    }
+    libxsmm_datatype get_type_in0() const {
+        return m_static_params->m_type_in0;
+    }
+    libxsmm_datatype get_type_in1() const {
+        return m_static_params->m_type_in1;
+    }
+    libxsmm_datatype get_type_out0() const {
+        return m_static_params->m_type_out0;
+    }
+    libxsmm_datatype get_type_exec() const {
+        return m_static_params->m_type_exec;
+    }
+
+private:
+    struct StaticParams : public StaticBaseParams {
+        StaticParams(const element::Type& in0_dtype,
+                     const element::Type& in1_dtype,
+                     dnnl::impl::cpu::aarch64::cpu_isa_t primitive_isa);
+        virtual ~StaticParams() = default;
+
+        bool operator==(const StaticParams& rhs) const;
+        bool operator!=(const StaticParams& rhs) const {
+            return !(*this == rhs);
+        }
+        size_t compute_hash(dnnl::impl::cpu::aarch64::cpu_isa_t aarch_isa);
+
+        dnnl::impl::cpu::aarch64::cpu_isa_t isa;
+        libxsmm_datatype m_type_in0;
+        libxsmm_datatype m_type_in1;
+        libxsmm_datatype m_type_out0;
+        libxsmm_datatype m_type_exec;
+        libxsmm_bitfield m_compile_flags;
+        bool m_prefetching_flags;
+    };
+    std::shared_ptr<StaticBaseParams> get_static_params() const override {
+        return m_static_params;
+    }
+
+    libxsmm_bitfield m_compile_flags{0};
+    std::shared_ptr<StaticParams> m_static_params{nullptr};
+};
+
+// The `update_kernel` method verifies that a compiled kernel is not nullptr.
+// However, the compiled kernel might be empty in cases if nothing is to be compiled (`Config.is_empty() == true`).
+// To cover this case, we wrap the `libxsmm_gemmfunction` in the separate structure which may contain empty
+// `libxsmm_gemmfunction`
+struct BrgemmTppCompiledKernel {
+    std::shared_ptr<libxsmm_gemmfunction> brgemm_kernel = nullptr;
+};
+
+class BrgemmKernelExecutor : public BrgemmBaseKernelExecutor,
```

Comment:
By the way, is there any differences of config/executor/kernel between aarch64?
Is there any chance to have only one kernel executor for x64 and aarc64? As far as I know, all aarch-dependent params are hidden in TPP functions/kernels (this is pro of libxsmm).

Reply:
If I understand correctly, you meant to reuse config/executor/kernel in TPP scope on x64 and aarch64. If yes I have created common tpp `BrgemmKernelConfig `and `BrgemmKernelExecutor ` This could be reused for both x64 and aarch64. Onednn backed kernel have differernt JIT config, could not be the same. I also applied the [comment ](https://github.com/openvinotoolkit/openvino/pull/28229#discussion_r1913104180). A super base brgemm config and executor created for tpp and onedenn backend. Thank you! Alexandra!